tests/smoke/smoke_test.go

```diff
@@ -1,13 +1,134 @@
 package smoke_test
 
 import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"time"
+
+	"github.com/accuknox/auto-policy-discovery/src/types"
+	"github.com/kubearmor/KubeArmor/tests/util"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	"github.com/rs/zerolog/log"
 )
 
+func getMatchDir(policy types.KubeArmorPolicy, sys string, str string) string {
+	if sys == "file" {
+		for i := range policy.Spec.File.MatchDirectories {
+			if policy.Spec.File.MatchDirectories[i].Dir == str {
+				value := str
+				return value
+			}
+		}
+	} else {
+		for i := range policy.Spec.Process.MatchDirectories {
+			if policy.Spec.Process.MatchDirectories[i].Dir == str {
+				value := str
+				return value
+			}
+		}
+	}
+	return ""
+}
+
+func getMatchPath(policy types.KubeArmorPolicy, sys string, str string) string {
+	if sys == "file" {
+		for i := range policy.Spec.File.MatchPaths {
+			if policy.Spec.File.MatchPaths[i].Path == str {
+				value := str
+				return value
+			}
+		}
+	} else {
+		for i := range policy.Spec.Process.MatchPaths {
+			if policy.Spec.Process.MatchPaths[i].Path == str {
+				value := str
+				return value
+			}
+		}
+	}
+	return ""
+}
+
+func checkPod(name string, ant string, ns string) {
+	pods, err := util.K8sGetPods(name, ns, []string{ant}, 60)
+	Expect(err).To(BeNil())
+	Expect(len(pods)).To(Equal(1))
+}
+
+var _ = BeforeSuite(func() {
+	// install discovery-engine
+	_, err := util.Kubectl(fmt.Sprintf("apply -f https://raw.githubusercontent.com/kubearmor/discovery-engine/dev/deployments/k8s/deployment.yaml"))
+	Expect(err).To(BeNil())
+	time.Sleep(20 * time.Second)
+
+	//install wordpress-mysql app
+	err = util.K8sApply([]string{"res/wordpress-mysql-deployment.yaml"})
+	Expect(err).To(BeNil())
+	time.Sleep(25 * time.Second)
```

Comment:
same here. This sleep should not be used.

Reply:
use `checkPod()` to check if the wordpress/mysql pods are deployed.

Reply:
@nyrahul this is covered in `BeforeEach()`, in `BeforeSuite` we install DE and wordpress/mysql pod and in `BeforeEach` we use `checkPod()` for checking if pods are deployed or not. https://github.com/accuknox/discovery-engine/pull/653/files#diff-d7590c6b861cadd571109c26d16100a5b65b48829ede89e2eb611126004fc701R93-R100